cmd/access/node_builder/staked_access_node_builder.go

```diff
@@ -89,8 +101,146 @@ func (builder *StakedAccessNodeBuilder) Initialize() error {
 	return nil
 }
 
-func (anb *StakedAccessNodeBuilder) Build() AccessNodeBuilder {
-	anb.FlowAccessNodeBuilder.Build()
+func (anb *StakedAccessNodeBuilder) Build() cmd.Node {
+	anb.
+		BuildConsensusFollower().
+		Module("collection node client", func(builder cmd.NodeBuilder, node *cmd.NodeConfig) error {
+			// collection node address is optional (if not specified, collection nodes will be chosen at random)
+			if strings.TrimSpace(anb.rpcConf.CollectionAddr) == "" {
+				node.Logger.Info().Msg("using a dynamic collection node address")
+				return nil
+			}
+
+			node.Logger.Info().
+				Str("collection_node", anb.rpcConf.CollectionAddr).
+				Msg("using the static collection node address")
+
+			collectionRPCConn, err := grpc.Dial(
+				anb.rpcConf.CollectionAddr,
+				grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(grpcutils.DefaultMaxMsgSize)),
+				grpc.WithInsecure(),
+				backend.WithClientUnaryInterceptor(anb.rpcConf.CollectionClientTimeout))
+			if err != nil {
+				return err
```

Comment:
general comment: better to prepend some contextual message to the error: `fmt.Errorf("could not dial collection node over grpc: %w", err)`